src/near-rpc-provider.ts

```diff
@@ -154,10 +162,75 @@ export class NearRpcProvider extends JsonRpcProvider {
     switch (method) {
       case 'getBlockNumber':
         const blockResponse = await this.send<BlockRpcResponse>('block', { finality: 'final' })
-
         return blockResponse.header.height
+      case 'getBalance':
+        return this._internalGetBalance(params)
+      case 'getBlock':
+        return this._internalGetBlock(params)
       default:
         return super.perform(method, params)
     }
   }
+
+  _getAddress(addressOrName: string | Promise<string>): Promise<string> {
+    return Promise.resolve(addressOrName)
+  }
+
+  async status() {
+    const statusResponse = await this.send<StatusRpcResponse>('status', {})
+    return statusResponse
+  }
+
+  async _getBlock(blockHashOrBlockTag: BlockTag | string | Promise<BlockTag | string>): Promise<Block> {
+    blockHashOrBlockTag = await blockHashOrBlockTag
+    const block = await this.perform('getBlock', { block_id: blockHashOrBlockTag })
+    return block
+  }
+
+  private async _internalGetBlock(params: Record<string, any>): Promise<Block> {
+    const blockResponse = await this.send<BlockRpcResponse>('block', { block_id: params.block_id })
+
+    return {
+      hash: blockResponse.header.hash,
+      parentHash: blockResponse.header.prev_hash,
+      number: blockResponse.header.height,
+      timestamp: blockResponse.header.timestamp,
+      // TODO: starting from here, are these values correct?
+      nonce: blockResponse.header.random_value,
+      difficulty: blockResponse.header.block_ordinal,
```

Comment:
The block ordinal is the amount of Blocks already existing on the Canonical Chain + 1